docs/input-manifest/providers/gcp.md

```diff
@@ -1,27 +1,278 @@
+## Compute and DNS example
 
-Furthermore, your project should enable a few API, namely
+```yaml
+providers:
+  gcp:
+    - name: gcp-1
+      gcpProject: sa-project-name
+      credentials: |
+      {
+         "type":"service_account",
+         "project_id":"project-claudie",
+         "private_key_id":"private_key_id",
+         "private_key":"private_key_string",
+         "client_email":"claudie@project-claudie-123456.iam.gserviceaccount.com",
+         "client_id":"109876543211234567890",
+         "auth_uri":"https://accounts.google.com/o/oauth2/auth",
+         "token_uri":"https://oauth2.googleapis.com/token",
+         "auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs",
+         "client_x509_cert_url":"https://www.googleapis.com/robot/v1/metadata/x509/claudie%40claudie-project-123456.iam.gserviceaccount.com"
+      }
+```
 
-- `Compute Engine API`
-- `Cloud DNS API`
+## Create GCP credentials
+### Prerequisites
+1. Install gcoud CLI on your machine by following [this guide](https://cloud.google.com/sdk/docs/install).
+2. Initialize gcloud CLI by following [this guide](https://cloud.google.com/sdk/docs/initializing).
+3. Authorize cloud CLI by following [this guide](https://cloud.google.com/sdk/docs/authorizing)
 
-> when project will be used for Loadbalancers DNS
+### Creating GCP credentials for Claudie
 
-## DNS requirements
+1. Create a GCP project:
+```bash
+gcloud projects create claudie-project
+```
 
-If your GCP provider will be used for DNS, you need to manually
+2. Set the current project to claudie-project:
+```bash
+gcloud config set project claudie-project
+```
 
-- [set up dns zone](https://cloud.google.com/dns/docs/zones)
-- [update domain name server](https://cloud.google.com/dns/docs/update-name-servers)
+3. Attach billing account to your project:
+```bash
+gcloud alpha billing accounts projects link claudie-project (--account-id=ACCOUNT_ID | --billing-account=ACCOUNT_ID)
+```
 
-since Claudie does not support their dynamic creation.
+4. Enable Compute Engine API and Cloud DNS API:
+```bash
+{
+  gcloud services enable compute.googleapis.com
+  gcloud services enable dns.googleapis.com
+}
+```
 
-## IAM policies required by Claudie
+5. Create a service account:
+```bash
+gcloud iam service-accounts create claudie-sa
+```
 
-- for infrastructure creation
-  - `roles/roles/compute.admin`
-  
-- for DNS
-  - `roles/dns.admin`
+6. Attach roles to the servcie account:
+```bash
+{
+  gcloud projects add-iam-policy-binding claudie-project --member=serviceAccount:claudie-sa@claudie-project.iam.gserviceaccount.com --role=roles/compute.admin
+  gcloud projects add-iam-policy-binding claudie-project --member=serviceAccount:claudie-sa@claudie-project.iam.gserviceaccount.com --role=roles/dns.admin
+}
+```
+
+7. Recover service account keys for claudie-sa:
+```bash
+gcloud iam service-accounts keys create claudie.json --iam-account=claudie-sa@claudie-project.iam.gserviceaccount.com
```

Comment:
Do you think we should add a manual step where the key is extracted into a local file, and then deployed as a secret into the cluster?

Reply:
I've added a compromise here, without scripting and being overly complex, additional step that shows user how to replace the json in the provided template.